buildflow/core/credentials/_credentials.py

```diff
@@ -0,0 +1,13 @@
+from typing import TypeVar
+
+from buildflow.core.options.credentials_options import CredentialsOptions
+
+
+class Credentials:
+    """Container class for different types of credentials we support."""
+
+    def __init__(self, credentials_options: CredentialsOptions) -> None:
+        pass
+
+
+CredentialType = TypeVar("CredentialType", bound=Credentials)
```

Comment:
love this idea

Reply:
mind adding a TODO (can assign to @tanke) to update the other types to do something similar?

Reply:
filed an issue here: https://github.com/launchflow/buildflow/issues/219